mitosheet/mitosheet/public/v3/sheet_functions/bool_functions.py

```diff
@@ -123,6 +124,76 @@ def IF(condition: pd.Series, true_series: AnyPrimitiveOrSeriesInputType, false_s
     )
 
 
+@handle_sheet_function_errors
+def IFS(*argv: Optional[IfsInputType]) -> pd.Series:
+    """
+    {
+        "function": "IFS",
+        "description": "Returns the value of the first condition that is true. If no conditions are true, returns None.",
+        "search_terms": ["ifs", "if", "conditional", "and", "or"],
+        "examples": [
+            "IFS(height > 100, 'tall', height > 50, 'medium', height > 0, 'short')",
+        ],
+        "syntax": "IFS(boolean_condition_1, value_if_true, boolean_condition_2, value_if_true)",
+        "syntax_elements": [{
+                "element": "boolean_condition",
+                "description": "An expression or series that returns True or False values. Valid conditions for comparison include ==, !=, >, <, >=, <=."
+            },
+            {
+                "element": "value_if_true",
+                "description": "The value the function returns if condition is True."
+            }, ...
+        ]
+    }
+    """
+    # Go through all of the conditions. The first condition that is true for
+    # *each cell* should be selected. 
+    results = None
+    for index, condition in enumerate(argv):
+        # The args are alternating between the conditions and the values corresponding to them.
+        # If this is a series of booleans in the even index, we assume it is the condition.
+        # We skip the values in this loop because they're only used as a function of the conditions. 
+        if index % 2 == 0:
+            if isinstance(condition, pd.Series) and condition.dtype == bool:
+                # For each cell, check if the value is true.
+                # If it is, use the "true_series" to fill the value in the result series.
+                true_series = get_series_from_primitive_or_series(argv[index+1], condition.index)
+                # Add the new data into the series
+                new_series = true_series[condition]
+                if results is None:
+                    results = new_series
+                else:
+                    # Combine the series to fill in the None values in the final series
+                    results = results.combine_first(new_series)
+            # This is the "else" case for the IFS formula.
+            elif condition == True:
+                base_index = next(iter(s.index for s in argv if isinstance(s, pd.Series)), None)
+                if base_index is None:
+                    error = MitoError(
+                        'invalid_args_error',
+                        'IFS',
+                        f"IFS requires at least one boolean series condition.",
+                        error_modal=False
+                    )
```

Comment:
This feels like an unnecessary limitation. Will describe an implementation that might remove this in a second -- let me know what you think.